packages/expo-modules-core/ios/JSI/EXJavaScriptRuntime.mm

```diff
@@ -19,35 +19,43 @@
 using namespace facebook;
 
 @implementation EXJavaScriptRuntime {
-  std::shared_ptr<jsi::Runtime> _runtime;
+  jsi::Runtime *_runtime;
```

Comment:
Based on our slack conversation I assumed that you need to have raw pointer here, but now I see that it's not. You can keep olde code the only change you would need to make is to:
- `initWithRuntime` needs to take raw pointer(already done in this PR)
- `_runtime = std::shared_ptr<jsi::Runtime>(std::shared_ptr<jsi::Runtime>(), runtime)` in `initWithRuntime`

You can check https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr for an explanation
constructor (8)

Reply:
the shared_ptr usage is cool. thanks @wkozyra95 for sharing awesome c++ knowledge as always 🤩